File: PYTHON/Pertemuan 01/Rekening.py
import sys
sys.stdout.reconfigure(encoding='utf-8')
class Rekening:
    def __init__(self, pemilik: str, saldo_awal: float):
        self._pemilik = pemilik
        self._saldo = saldo_awal   # private → tidak bisa diakses langsung dari luar

    # "Pintu resmi" – saldo hanya bisa berubah melalui method berikut:
    def setor(self, jumlah: float):
        if jumlah > 0:
            self._saldo += jumlah
            print(f"[{self._pemilik}] Setor Rp{jumlah:.0f} → Saldo: Rp{self._saldo:.0f}")

    def tarik(self, jumlah: float):
        if 0 < jumlah <= self._saldo:
            self._saldo -= jumlah
            print(f"[{self._pemilik}] Tarik Rp{jumlah:.0f} → Saldo: Rp{self._saldo:.0f}")
        else:
            print(f"[{self._pemilik}] ⚠ Saldo tidak cukup!")

    @property
    def saldo(self) -> float:
        return self._saldo

    @property
    def pemilik(self) -> str:
        return self._pemilik

File: JAVA/Pertemuan 01/src/id/ac/uph/pbo/pertemuan01/Rekening.java
package id.ac.uph.pbo.pertemuan01;

public class Rekening {

    private String pemilik;
    private double saldo;   // private → tidak bisa diakses langsung dari luar

    public Rekening(String pemilik, double saldoAwal) {
        this.pemilik = pemilik;
        this.saldo   = saldoAwal;
    }

    // "Pintu resmi" – saldo hanya bisa berubah melalui method berikut:
    public void setor(double jumlah) {
        if (jumlah > 0) {
            saldo += jumlah;
            System.out.printf("[%s] Setor Rp%.0f → Saldo: Rp%.0f%n",
                    pemilik, jumlah, saldo);
        }
    }

    public void tarik(double jumlah) {
        if (jumlah > 0 && jumlah <= saldo) {
            saldo -= jumlah;
            System.out.printf("[%s] Tarik Rp%.0f → Saldo: Rp%.0f%n",
                    pemilik, jumlah, saldo);
        } else {
            System.out.println("[" + pemilik + "] ⚠ Saldo tidak cukup!");
        }
    }

    public double getSaldo() { return saldo; }
    public String getPemilik() { return pemilik; }
}
